src/utils/feeDistributor.js

```diff
@@ -1,66 +1,92 @@
+import { ethers } from "ethers";
 import * as feeDistributorABI from "./abis/feeDistributorABI";
-import {getAddressByChainIdKey} from "../utils/address/address";
+import { getAddressByChainIdKey } from "../utils/address/address";
+import { prepareWriteContract, readContract, writeContract } from "@wagmi/core";
+import { getGasFeeEstimate } from "./web3";
 
-export const getTimeCursor = async(userAddress, provider) => {
+export const getTimeCursor = async (userAddress) => {
   try {
-    const contract = new ethers.Contract(
-      getAddressByChainIdKey(process.env.VE_SUPPORTED_CHAINID, "veFeeDistributor"),
-      feeDistributorABI.default, 
-      provider
-    );
-    const weekCursor = await contract.time_cursor_of(userAddress);
+    const weekCursor = await readContract({
+      address: getAddressByChainIdKey(
+        import.meta.env.VITE_VE_SUPPORTED_CHAINID,
+        "veFeeDistributor"
+      ),
+      args: [userAddress],
+      abi: feeDistributorABI.default,
+      functionName: "time_cursor_of",
+    });
     return parseInt(BigInt(weekCursor));
   } catch (error) {
-    console.log(error)
+    console.log(error);
     throw error;
   }
-}
+};
 
-export const getUserEpoch = async(userAddress, provider) => {
+export const getUserEpoch = async (userAddress, provider) => {
   try {
-    const contract = new ethers.Contract(
-      getAddressByChainIdKey(process.env.VE_SUPPORTED_CHAINID, "veFeeDistributor"),
-      feeDistributorABI.default, 
-      provider
-    );
-    const userEpoch = await contract.user_epoch_of(userAddress);
+    const userEpoch = await readContract({
+      address: getAddressByChainIdKey(
+        import.meta.env.VITE_VE_SUPPORTED_CHAINID,
+        "veFeeDistributor"
+      ),
+      args: [userAddress],
+      abi: feeDistributorABI.default,
+      functionName: "user_epoch_of",
+    });
     return parseInt(BigInt(userEpoch));
   } catch (error) {
-    console.log(error)
+    console.log(error);
     throw error;
   }
-}
+};
 
-export const getLastTokenTime = async(provider) => {
+export const getLastTokenTime = async () => {
   try {
-    const contract = new ethers.Contract(
-      getAddressByChainIdKey(process.env.VE_SUPPORTED_CHAINID, "veFeeDistributor"),
-      feeDistributorABI.default, 
-      provider
-    );
-    const calcGasLimit = await contract.estimateGas.last_token_time()
-    const lastTokenTime = await contract.last_token_time({gasLimit:BigInt(calcGasLimit) + BigInt(10000)});
+    const lastTokenTime = await readContract({
+      address: getAddressByChainIdKey(
+        import.meta.env.VITE_VE_SUPPORTED_CHAINID,
+        "veFeeDistributor"
+      ),
+      args: [],
+      abi: feeDistributorABI.default,
+      functionName: "last_token_time",
+    });
     return parseInt(BigInt(lastTokenTime));
   } catch (error) {
-    console.log(error)
+    console.log(error);
     throw error;
   }
-}
+};
 
-export async function claim(userAddress, signer) {
+export async function claim(userAddress) {
```

Comment:
Did not test latest implementation of claim functions inside in-app/test environment, only goerli.
Will continue with testing throughout this week.